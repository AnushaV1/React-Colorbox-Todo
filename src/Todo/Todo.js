import React from "react";

const Todo = ({task = "model todo", id = "1", remove}) => {
    const handleDelete = () => remove(id);
return (
    <div>
        <li>{task}</li>
        <button onClick={handleDelete}>X</button>
    </div>
)

}

export default Todo;