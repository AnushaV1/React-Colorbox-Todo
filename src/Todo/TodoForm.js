import React, {useState} from "react";
import { v4 as uuid } from 'uuid';
const TodoForm = ({ addTodo }) => {
    const [task, setTask] = useState("");

    const handleChange = (e) => {
        setTask(e.target.value);
        };

    
    const handleSubmit = (e) => {
        e.preventDefault();
        addTodo({task, id: uuid()});
        setTask("");
    }


    return (
    <form onSubmit={handleSubmit}>
    <label htmlFor='task'>Add Task</label>
    <input type="text"
        name="task"
        placeholder="Enter Todo item"
        id="task"
        onChange ={handleChange}
        value = {task}
    />
    <button>Add To List</button>
    </form>

    )
}

export default TodoForm;