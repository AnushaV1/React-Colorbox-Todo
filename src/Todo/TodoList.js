import React, {useState} from "react";
import Todo from "./Todo";
import TodoForm from './TodoForm'
const TodoList = () => {

    const [todos, setTodos] = useState([]);
    
    const addTodo = (newTodo) => {
        setTodos(todos => [...todos, newTodo]);
    }

    const remove = id => {
        setTodos(todos => todos.filter(todo => todo.id !==id))
    }

    return (
    <>
        <TodoForm addTodo={addTodo}/>
    <div>    <ul>
    {todos.map(todo => (
    <Todo
        remove={remove}
        key={todo.id}
        id={todo.id}
        task={todo.task}
    />
    ))
    }
    </ul>
    </div>
    </>

    )
}

export default TodoList;