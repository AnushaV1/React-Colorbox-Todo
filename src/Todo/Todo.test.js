import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Todo from './Todo';

it('renders  without crashing', function() {
  render(<Todo />);
})

it('matching snapshot', function() {
    const {asFragment} = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

it('delete function on button click ', function() {
    const deleteMock = jest.fn();
    const { getByText } =  render(<Todo remove={deleteMock} />);
    const deleteBtn = getByText('X');
    fireEvent.click(deleteBtn);
    expect(deleteMock).toHaveBeenCalled();
});