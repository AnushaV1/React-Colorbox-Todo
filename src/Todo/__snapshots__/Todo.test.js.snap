// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matching snapshot 1`] = `
<DocumentFragment>
  <div>
    <li>
      model todo
    </li>
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
